### ISSUE TITLE
[Discussion] Remove defined('TESTSUITE') from codebase

### ISSUE THREAD

**Issue by williamdes** [phpmyadmin/phpmyadmin#15247]
### We can do better than that

- Remove all hacks `if (defined('TESTSUITE'))`
- Implement a session handling system for the testsuite

Please post your thoughts on why we could not remove `defined('TESTSUITE')`

**Comment by mauriciofauth** [comment]
I agree with you. We can definitely find a solution that doesn't require this constant.

**Comment by williamdes** [comment]
Needs to be keeped or detected for `ErrorHandler.php` & `Config.php` and `DatabaseInterface.php` for `DbiDummy`

**Comment by williamdes** [comment]
- [ ] `libraries/classes/Core.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Core.php:        if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Core.php:        if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Relation.php:        if (defined('TESTSUITE')`
- [ ] `libraries/classes/Response.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Response.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Html/MySQLDocumentation.php:        if (! defined('TESTSUITE') && @file_exists(ROOT_PATH . 'doc/html/index.html')) {`
- [ ] `libraries/classes/DatabaseInterface.php:        if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Session.php:        if (session_status() === PHP_SESSION_ACTIVE && ! defined('TESTSUITE')) {`
- [ ] `libraries/classes/OutputBuffering.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/InsertEdit.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Util.php:        if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/ErrorHandler.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/AuthenticationPlugin.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/AuthenticationPlugin.php:            if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/AuthenticationPlugin.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationConfig.php:            if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationConfig.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationCookie.php:            if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationCookie.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationCookie.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationCookie.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationCookie.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationHttp.php:            if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationHttp.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationSignon.php:        if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationSignon.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationSignon.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationSignon.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationSignon.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Auth/AuthenticationSignon.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Export/ExportSql.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Plugins/Export/ExportSql.php:            if (! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Config.php:        if ($cookie_path !== null && ! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Config.php:        if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Config.php:            if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Config.php:        if (isset($temp_dir[$name]) && ! defined('TESTSUITE')) {`
- [ ] `libraries/classes/Header.php:        if (defined('TESTSUITE')) {`
- [ ] `libraries/classes/Config/FormDisplayTemplate.php:        if (defined('TESTSUITE')) {`
- [ ] `test/bootstrap-static.php:define('TESTSUITE', true);`
- [ ] `test/bootstrap-dist.php:    define('TESTSUITE', 1);`

**Comment by williamdes** [comment]
Not so much remaining in 2024

**Comment by williamdes** [comment]
2026 Update:
```console
$ grep -r -F "defined('TESTSUITE')"
src/Sql.php:        if (! defined('TESTSUITE')) {
src/Import/Ajax.php:        $uploadId = ! defined('TESTSUITE') ? uniqid() : 'abc1234567890';
src/Import/Ajax.php:        return ! defined('TESTSUITE') && function_exists('uploadprogress_get_info');
src/Import/Ajax.php:        return ! defined('TESTSUITE') && ini_get('session.upload_progress.enabled') === '1';
src/Html/MySQLDocumentation.php:        if (! defined('TESTSUITE') && @file_exists(ROOT_PATH . 'docs/html/index.html')) {
src/Error/ErrorHandler.php:        if (defined('TESTSUITE')) {
src/Plugins/AuthenticationPlugin.php:            if (! defined('TESTSUITE')) {
src/Plugins/Auth/AuthenticationSignon.php:            if (! defined('TESTSUITE')) {
src/Config.php:        if (defined('TESTSUITE')) {
tests/bootstrap-static.php:if (! defined('TESTSUITE')) {
```
